src/ServiceManagement/Compute/Commands.ServiceManagement/Model/RoleConfiguration.cs

```diff
@@ -104,8 +105,18 @@ internal XElement Serialize()
             {
                 XElement certificateElement = new XElement(this.ns + "Certificate");
                 certificateElement.SetAttributeValue("name", certificate.Key);
-                certificateElement.SetAttributeValue("thumbprint", certificate.Value.Thumbprint);
-                certificateElement.SetAttributeValue("thumbprintAlgorithm", certificate.Value.ThumbprintAlgorithm);
+                if (certificate.Value.Thumbprint != null)
+                {
+                    certificateElement.SetAttributeValue("thumbprint", certificate.Value.Thumbprint);
+                }
+                if (certificate.Value.ThumbprintAlgorithm != null)
+                {
+                    certificateElement.SetAttributeValue("thumbprintAlgorithm", certificate.Value.ThumbprintAlgorithm);
```

Comment:
Would it be better to define a constant string instead of using hard coded string literals?

Reply:
I was following the existing pattern of the file. There are no constants at all.